applications/GeoMechanicsApplication/custom_elements/line_interface_element.cpp

```diff
@@ -0,0 +1,277 @@
+{
+    return Create(NewId, this->GetGeometry().Create(rNodes), pProperties);
+}
+
+Element::Pointer LineInterfaceElement::Create(IndexType               NewId,
+                                              GeometryType::Pointer   pGeometry,
+                                              PropertiesType::Pointer pProperties) const
+{
+    return {make_intrusive<LineInterfaceElement>(NewId, pGeometry, pProperties)};
+}
+
+void LineInterfaceElement::GetDofList(DofsVectorType& rElementalDofList, const ProcessInfo&) const
+{
+    rElementalDofList = GetDofs();
+}
+
+void LineInterfaceElement::Initialize(const ProcessInfo& rCurrentProcessInfo)
+{
+    BaseType::Initialize(rCurrentProcessInfo);
+
+    const auto unused_shape_function_values = Vector{};
+    for (auto i = std::size_t{0}; i < mIntegrationScheme->GetNumberOfIntegrationPoints(); ++i) {
+        mConstitutiveLaws.push_back(GetProperties()[CONSTITUTIVE_LAW]->Clone());
+        mConstitutiveLaws.back()->InitializeMaterial(GetProperties(), GetGeometry(), unused_shape_function_values);
+    }
+}
+
+Element::DofsVectorType LineInterfaceElement::GetDofs() const
+{
+    const auto no_Pw_geometry = Geometry<Node>{};
+    return Geo::DofUtilities::ExtractUPwDofsFromNodes(GetGeometry(), no_Pw_geometry,
+                                                      GetGeometry().WorkingSpaceDimension());
+}
+
+std::vector<Matrix> LineInterfaceElement::CalculateLocalBMatricesAtIntegrationPoints() const
+{
+    const auto& r_integration_points = mIntegrationScheme->GetIntegrationPoints();
+    const auto  shape_function_values_at_integration_points =
+        GeoElementUtilities::EvaluateShapeFunctionsAtIntegrationPoints(r_integration_points, GetGeometry());
+
+    auto result = std::vector<Matrix>{};
+    auto calculate_local_b_matrix = [&r_geometry = GetGeometry(), p_policy = mStressStatePolicy.get()](
+                                        const auto& rShapeFunctionValuesAtIntegrationPoint,
+                                        const auto& rIntegrationPoint) {
+        const auto dummy_gradients = Matrix{};
+        return Matrix{prod(
+            GeometryUtilities::Calculate2DRotationMatrixForLineGeometry(r_geometry, rIntegrationPoint),
+            p_policy->CalculateBMatrix(dummy_gradients, rShapeFunctionValuesAtIntegrationPoint, r_geometry))};
+    };
+    std::transform(shape_function_values_at_integration_points.begin(),
+                   shape_function_values_at_integration_points.end(), r_integration_points.begin(),
+                   std::back_inserter(result), calculate_local_b_matrix);
+
+    return result;
+}
+
+std::vector<double> LineInterfaceElement::CalculateIntegrationCoefficients() const
+{
+    auto calculate_integration_coefficient = [p_policy = mStressStatePolicy.get()](
+                                                 const Geo::IntegrationPointType& rIntegrationPoint,
+                                                 double DetJ, const Geometry<Node>& rGeometry) {
```

Comment:
We can infer the types here:
```suggestion
                                                 const auto& rIntegrationPoint,
                                                 auto DetJ, const auto& rGeometry) {
```